### PR TITLE
Runtime: Remove support_locality_update_on_eds_cluster_endpointss

### PR DESCRIPTION
**Pull request by vikaschoudhary16**
Signed-off-by: Vikas Choudhary <choudharyvikas16@gmail.com>

Commit Message:
Additional Description:
Risk Level: low
Testing:
Docs Changes:
Fixes: https://github.com/envoyproxy/envoy/issues/23603

### CODE REVIEW

File: source/common/runtime/runtime_features.cc [hunk 1]
```diff
@@ -67,7 +67,6 @@ RUNTIME_GUARD(envoy_reloadable_features_postpone_h3_client_connect_to_next_loop)
 RUNTIME_GUARD(envoy_reloadable_features_quic_defer_send_in_response_to_packet);
 RUNTIME_GUARD(envoy_reloadable_features_skip_delay_close);
 RUNTIME_GUARD(envoy_reloadable_features_strict_check_on_ipv4_compat);
-RUNTIME_GUARD(envoy_reloadable_features_support_locality_update_on_eds_cluster_endpoints);
```
Comment (suniltheta) [inline source/common/runtime/runtime_features.cc:70]: Do you want to mention this change under "Removed config or runtime" changelog ?

File: source/common/upstream/upstream_impl.cc [hunk 2]
```diff
@@ -1872,14 +1872,10 @@ bool BaseDynamicClusterImpl::updateDynamicHostList(
...
     bool locality_changed = false;
-    if (Runtime::runtimeFeatureEnabled(
-            "envoy.reloadable_features.support_locality_update_on_eds_cluster_endpoints")) {
-      locality_changed =
-          (existing_host_found &&
-           (!LocalityEqualTo()(host->locality(), existing_host->second->locality())));
-      if (locality_changed) {
-        hosts_with_updated_locality_for_current_priority.emplace(existing_host->first);
-      }
+    locality_changed = (existing_host_found &&
```
Comment (wbpcode) [inline source/common/upstream/upstream_impl.cc:1875]: `const bool locality_changed = ....`

**Review by wbpcode — COMMENTED** [review]
Thanks. LGTM overall. Only a minor comment.

**Comment by vikaschoudhary16** [comment]
closing because of https://github.com/envoyproxy/envoy/pull/23673

**Comment by wbpcode** [comment]
Still thanks for this work. Thanks. 🌹 cc @vikaschoudhary16

**Pull request closed without merging**